### ISSUE TITLE
Training accuracy cannot go beyond 0.8

### ISSUE THREAD

**Issue by zguo008** [cardwing/Codes-for-Lane-Detection#107]
Hi,

I spent days re-training SCNN, the training accuracy cannot go beyond 80%.

Since theres no tensorboard summary written in the code, may I know how is your training accuracy?

Will it go up as high as Lanenet's accuracy like 95%?

Best,

ZX

**Comment by zguo008** [comment]
I wrote a tf summary in the code, so my tensorboard now looks like:
![image](https://user-images.githubusercontent.com/44756093/57761074-57b29380-772f-11e9-9ed5-13fa0233e263.png)
Is this normal?

**Comment by cardwing** [comment]
The training accuracy should be around 95%.

**Comment by zguo008** [comment]
Okay, so my training is not sucessful by now. Do you have any idea why my training accracy is fluctuating? Is this normal?

**Comment by cardwing** [comment]
What is your keep prob in dropout? If you use the default setting (keep_prob = 0.9), the training accuracy should be around 95%.

**Comment by zguo008** [comment]
I trained it with UNET as backbone, I kept the dropout from UNET backbone as well, so maybe this is the reason I think..There are 3 dropouts now. Have u tried batch norm before? If yes how's the performance?

**Comment by cardwing** [comment]
I have not met the fluctuation during training. But the different model may account for the fluctuation since some hyperparameters may need to be modified accordingly.

**Comment by sumitbinnani** [comment]
How is the accuracy calculated?

Or more precisely, what is the value of `alpha` in the following equation? 
```final accuracy = alpha * accuracy + (1-alpha) * accuracy```

**Comment by zguo008** [comment]
@cardwing I think the too much dropout causes the fluctuation. I removed some dropout layers and the fluctuation is not that severe now:) since BN is implemented here and it can have some regularization effect so too much dropout is unnecessary. will update you on further result.

**Comment by zguo008** [comment]
@sumitbinnani I am not that experienced in this field loll, so I am not familiar with the equation( accuracy*alpha - accuracy*alpha=0, so the equation is effectively, final acc=acc?) 
I didnt modify the accuracy calculation process here, so you can refer to train.py. Also there is a evaluation script at https://github.com/XingangPan/SCNN showing the F-measure calculating process, you can have a look:)

**Comment by zguo008** [comment]
The fluctuation was indeed caused by the dropout term. Now the training acc increases by 10% roughly.

I am thinking embellish the message passing layer at the end of the model at the 5 channel (batch,H,W,5) part. But I am a bit confused about the weight  initialization. 
Should I change to something like this?
w3 = tf.get_variable('W3', [9, 1, 5, 5],
                                 initializer=tf.random_normal_initializer(0, math.sqrt(2.0 / (9 * 5 * 5 * 5))))

Original code's weight's size is [9,1,128,128], I wonder what does the second 128 dimension means? shouldnt it be [9,1,128]?

Thank you:)

**Comment by cardwing** [comment]
Yes. And the second 128 is the output dimension. Typically, the shape of a conv layer's weight is [kernel_size, kernel_size, input_dim, output_dim].

**Comment by zguo008** [comment]
Thank you:) have a nice day!
